src/errors.rs

```diff
@@ -293,3 +294,21 @@ pub(crate) use err_opt;
 pub(crate) use format_err;
 pub(crate) use unimplemented_err;
 pub(crate) use unsupported_err;
+
+#[cfg(test)]
+mod tests {
+    /// Check the size of the error enum
+    ///
+    /// Because clippy will start throwing warning if an enum gets above 128, we'd like to keep the
+    /// size of the `Error` enum lower than that limit with some headroom to be wrapped by a
+    /// downstream crate.
+    ///
+    /// If this test triggers, you should consider Box'ing the offending member.
+    ///
+    /// See: <https://rust-lang.github.io/rust-clippy/master/index.html#result_large_err>
+    #[cfg(target_pointer_width = "64")]
+    #[test]
+    fn size_of_error() {
+        assert_eq!(core::mem::size_of::<super::Error>(), 80);
```

Comment:
I wonder if this shouldn't be something like:

```suggestion
        assert!(core::mem::size_of::<super::Error>() <= 80, "Error needs a diet!");
```

That is if the error is smaller than 80 we don't care about the exact value. (The error message is just a sample :wink: )

Reply:
@wiktor-k Sorry, race condition :scream: 

... let's do these changes once the test complains, in the future ;)

Reply:
Oh, that's a good idea too! Also it doesn't seem like it would go down by accident given the amount of variants there. Okay, thanks! :wave: